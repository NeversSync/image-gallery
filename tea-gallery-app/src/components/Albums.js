import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import {
  Route,
  Link,
} from 'react-router-dom';
import albumApi from '../data';
// import AddAlbum from './components/AddAlbum';
import AlbumDetail from './AlbumDetail';

class Albums extends Component {

  // static propTypes = {
  //   images: PropTypes.array,
  // }

  constructor(props) {
    super(props);

    this.state = {
      albums: []
    };
  }
  
  componentDidMount() {
    albumApi.get()
      .then(albums => {
        this.setState({
          albums
        });
      });
  }

  render() {
    const { match } = this.props;
    const { albums } = this.state;
    return (
      <div>
        <div className="albums-view">
          {albums.map(album =>
            <button
              key={album.name}
              onClick={() => this.setState({ album })}
            >
              <Link to={`/albums/${album._id}`}>{album.name}</Link>
            </button>)}
        </div>
        <Route path={`${match.url}/:albumId`} component={AlbumDetail}></Route> 
      </div>
    );
  }
}

export default Albums;
