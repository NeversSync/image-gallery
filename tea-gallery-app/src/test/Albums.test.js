import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Albums from '../components/Albums';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
  }
];

const match = {
  url: 'google.com'
};


describe.skip('Albums', () => {
  it('renders the Albums view', () => {
    const component = renderer.create(<MemoryRouter initialEntries={[ '/albums', { pathname: '/albums' } ]}
  initialIndex={1}>
    <Albums match={match} images={images}/>
    </MemoryRouter>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});