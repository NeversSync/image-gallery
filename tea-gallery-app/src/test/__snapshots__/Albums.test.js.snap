// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders the Albums view 1`] = `
<div>
  <div
    className="albums-view"
  >
    <h3>
      Albums
    </h3>
    <form
      className="add-item"
      onSubmit={[Function]}
    >
      <label>
        Add an album
        <input
          name="name"
          onChange={[Function]}
          placeholder="album name"
        />
      </label>
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</div>
`;
